controllers/admin/AdminImportController.php

```diff
@@ -3762,44 +3797,66 @@ public function storeContactImportOne($info, $shop_is_feature_active, $regenerat
                     !$validateOnly && // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
                     $state->add()) {
                     $store->id_state = (int)$state->id;
                 } else {
                     if (!$validateOnly) {
                         $this->errors[] = sprintf($this->trans('%s cannot be saved', array(), 'Admin.Advparameters.Notification'), $state->name);
                     }
-                    if ($field_error !== true || isset($lang_field_error) && $lang_field_error !== true) {
-                        $this->errors[] = ($field_error !== true ? $field_error : '').(isset($lang_field_error) && $lang_field_error !== true ? $lang_field_error : '').
+                    if ($fieldsValidationResult !== true || isset($langFieldsValidationResult) && $langFieldsValidationResult !== true) {
+                        $this->errors[] = ($fieldsValidationResult !== true ? $fieldsValidationResult : '').(isset($langFieldsValidationResult) && $langFieldsValidationResult !== true ? $langFieldsValidationResult : '').
                             Db::getInstance()->getMsgError();
                     }
                 }
             }
         }
 
-        $res = false;
-        if (($field_error = $store->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-            ($lang_field_error = $store->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true) {
-            if ($store->id && $store->storeExists($store->id)) {
-                $res = $validateOnly ? $validateOnly : $store->update();
+        $fieldsValidationResult     = $store->validateFields(UNFRIENDLY_ERROR, true);
+        $langFieldsValidationResult = $store->validateFieldsLang(UNFRIENDLY_ERROR, true);
+
+        // If errors, log them and stop execution
+        if (true !== $fieldsValidationResult || true !== $langFieldsValidationResult) {
+            $errorGenericMessage = $this->trans('Store is invalid', [], 'Admin.Advparameters.Notification');
+            $this->errors[]      = $errorGenericMessage . ' (' . $store->name . ')';
+
+            $errorDetails = '';
+            if ($fieldsValidationResult !== true) {
+                $errorDetails .= $fieldsValidationResult;
             }
-            $store->force_id = (bool)$force_ids;
-            if (!$res) {
-                $res = $validateOnly ? $validateOnly : $store->add();
+            if ($langFieldsValidationResult !== true) {
+                $errorDetails .= $langFieldsValidationResult;
             }
+            $this->errors[] = $errorDetails;
 
-            if (!$res) {
-                $this->errors[] = Db::getInstance()->getMsgError().' '.sprintf(
+            return;
+        }
+
+        if ($validateOnly) {
+            return;
+        }
+
+        $res = false;
+        if ($store->storeExists($store->id)) {
+            $res = $store->update();
```

Comment:
Would `save()` help you? It handles both `update()` and `add()` whether it exists or not.